### PR TITLE
Change TJvDBComboBox search to be case sensitive

### PR DESCRIPTION
**Pull request by MHumm**
Made search Key case sensitive by default but additionally provided a property to control this.
Fix for Mantis issue 6361.
[http://issuetracker.delphi-jedi.org/view.php?id=6361](http://issuetracker.delphi-jedi.org/view.php?id=6361)

### CODE REVIEW

File: jvcl/run/JvDBCombobox.pas [hunk 1]
```diff
@@ -188,6 +190,7 @@   TJvCustomDBComboBox = class(TJvCustomComboBox, IJvDataControl)
     property Text;
     property UpdateFieldImmediatelly: Boolean read FUpdateFieldImmediatelly write FUpdateFieldImmediatelly default False;
     property PreserveItemSelectionOnInsert: Boolean read FPreserveItemSelectionOnInsert write FPreserveItemSelectionOnInsert default False;
+    property CaseSensitiveSearch: Boolean read GetCaseSensitiveSearchValue write SetCaseSensitiveSearchValue default true;
```
Comment (ronaldhoek) [inline jvcl/run/JvDBCombobox.pas]: Change too
`property CaseSensitiveSearch: Boolean ... default false;`

File: jvcl/run/JvDBCombobox.pas [hunk 2]
```diff
@@ -302,6 +306,7 @@ constructor TJvCustomDBComboBox.Create(AOwner: TComponent);
 
   FListSettings := TJvDBComboBoxListSettings.Create(Self);
   FValues := TStringList.Create;
+  FValues.CaseSensitive := True;
```
Comment (ronaldhoek) [inline jvcl/run/JvDBCombobox.pas]: Remove (so `FValues.CaseSensitive` remains `false`)

**Comment by ronaldhoek** [comment]
I would suggest the default behavior too remain case insensitive. This is mostly de default behavior of text searches... (See code review suggestions)

**Comment by MHumm** [comment]
Created a new commit with your requested changes. I hope it can be merged now.

**Comment by ronaldhoek** [comment]
The code works as to having 'case sensitive key values'  (zie 'Values' property).
But when `EnableValues` is `false` the combo box still works case insensitive regarding the stored data.

Two solutions:
- change property name to 'CaseSensitiveValues', therefore making it clear it only functions when using the `Values` property data when `EnableValues` is `true`  (= default)
- also, make it work when `EnableValues` is `false` (probably need to do something with the `Items` data..)

**Comment by MHumm** [comment]
As you can see I opted for your first suggestion. I hope it is good enough for now.

File: jvcl/run/JvDBCombobox.pas [hunk 3]
```diff
@@ -143,6 +143,8 @@   TJvCustomDBComboBox = class(TJvCustomComboBox, IJvDataControl)
     procedure CMGetDataLink(var Msg: TMessage); message CM_GETDATALINK;
     procedure WMPaint(var Msg: TWMPaint); message WM_PAINT;
     procedure SetListSettings(const Value: TJvDBComboBoxListSettings);
+    procedure SetCaseSensitiveSearchValue(Value: Boolean);
+    function GetCaseSensitiveSearchValue: Boolean;
```
Comment (obones) [inline jvcl/run/JvDBCombobox.pas]: Getters and Setters must be named after their associated property. In this case this means `SetCaseSensitiveValues` and `GetCaseSensitiveValues`

File: jvcl/run/JvDBCombobox.pas [hunk 4]
```diff
@@ -240,6 +243,7 @@   TJvDBComboBox = class(TJvCustomDBComboBox)
     property UpdateFieldImmediatelly;
     property Values;
     property Visible;
+    property CaseSensitiveSearch;
```
Comment (obones) [inline jvcl/run/JvDBCombobox.pas]: This cannot compile, it's not named the same as in the ancestor class

File: jvcl/run/JvDBCombobox.pas [hunk 5]
```diff
@@ -410,6 +419,11 @@ procedure TJvCustomDBComboBox.SetComboText(const Value: string);
   end;
 end;
 
+function TJvCustomDBComboBox.GetCaseSensitiveSearchValue: Boolean;
+begin
+  result := FValues.CaseSensitive;
```
Comment (obones) [inline jvcl/run/JvDBCombobox.pas]: `Result` takes a leading capital letter

**Comment by MHumm** [comment]
Renamed the getters and setters as requested. Please merge in now if good enough now.

**Comment by obones** [comment]
There are two remarks remaining

**Comment by ronaldhoek** [comment]
> There are two remarks remaining

When these are both resolved (I made the change locally) the code functions as intended.

**Comment by MHumm** [comment]
Ok, fixed both issues, so I hope this can be pulled in now.

**Comment by ronaldhoek** [comment]
Tested using applied changes - from my part OK to merge @obones

**Pull request merged**